src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs

```diff
@@ -114,5 +128,108 @@ internal static bool TryGetPathOfFileAbove(string file, DirectoryInfo startingDi
 
             return false;
         }
+
+        /// <summary>
+        /// Parses the <c>msbuild-sdks</c> section of the specified JSON string.
+        /// </summary>
+        /// <param name="json">The JSON to parse as a string.</param>
+        /// <returns>A <see cref="Dictionary{TKey, TValue}" /> containing MSBuild project SDK versions if any were found, otherwise <c>null</c>.</returns>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Dictionary<string, string> ParseMSBuildSdkVersionsFromJson(string json)
```

Comment:
If I understand correctly, before this change, it was assured that if the `msbuild-sdks` property was read successfully that the entire global.json was well-formed; however, with this change, this assurance no longer exists.  For example, the top-level JSON object could be lacking a closing `}`.  I presume that such an error would be now caught by some later reader.

Is this a concern?

Reply:
Yes the parsing is _slightly_ more permissive at the cost of being faster.  Invalid JSON before and after the `msbuild-sdks` section will still produce an error since I'm calling `.Skip()` to read all of the top-level elements.  Invalid **values** in the msbuild-sdks section are ignored now but invalid JSON is not.  In my opinion, its okay for this parsing logic to be less strict and I have added unit tests to verify what is and isn't going to give an error.